### DISCUSSION TITLE
Access  to query urls params in boot

### DISCUSSION THREAD

**Discussion by roylans** [quasarframework/quasar#10646] [General - Components / Directives / etc] [answered]
Hi

I need to access to a query url param in boot

![image](https://user-images.githubusercontent.com/16232825/132991081-7ecc33b2-f077-4d16-8e27-aafd1a9636c3.png)

And this my boot

`import { boot } from 'quasar/wrappers'

export default boot(async ({ router }) => {
  console.log(router.currentRoute.query)
})`

but in the console it tells me that it is undefined

![image](https://user-images.githubusercontent.com/16232825/132991168-edc2e73c-5966-4c9d-89d2-bf5add278a83.png)

What am I doing wrong?

Thnks

**Comment by hawkeye64** [comment]
try `console.log(location)`

**Comment by ontwikkelfabriek** [comment, marked as answer]
@roylans 

Here you can see what objects you can pass to the boot function.:
https://quasar.dev/quasar-cli/boot-files#anatomy-of-a-boot-file

What about using the `urlPath` instead of the `router` object to access the query params? See here:
https://quasar.dev/quasar-cli/boot-files#redirecting-to-another-page

**Comment by sjoshid** [comment]
I dont agree the marked answer is the _right_ answer. Sure `urlPath` can get you there but I hate parsing key/value pairs. 

`currentRoute` is a `Ref`. So you need to `router.currentRoute.value.query` to fix the `undefined` error. 

But, you still won't be able to get hold of query param. It seems it's too early to get that in boot file because `query` is empty.

**Reply by yusufkandemir** [reply]
As you said, it's too early to use vue router stuff in there. So, while it's not exactly as easy as using the router, it's the only logically possible solution(_also works fine in SSR_), as an alternative to using `window.location`.

**Comment by DarkLite1** [comment]
Ran into the same issue of trying to acces ` router.currentRoute.value.params.country` within a `boot` file. The funny thing is that `console.log(router)` does return all properties, but when accessing our `params`, it says `Undefined`. 

Will have a look at the suggestion of using `UrlPath` instead. Maybe this could be updated in the [boot file docs](https://quasar.dev/quasar-cli-webpack/boot-files#usage-of-boot-files), that certain properties are not available.
